helper/properties.go

```diff
@@ -36,10 +36,17 @@ type Properties struct {
 }
 
 func (p Properties) Execute() (map[string]string, error) {
-	b, ok, err := bindings.ResolveOne(p.Bindings, bindings.OfType("Dynatrace"))
+	b, bindingFound, err := bindings.ResolveOne(p.Bindings, bindings.WithName("Dynatrace"))
```

Comment:
@c0d1ngm0nk3y I think we might need to give some freedom for cases of existing bindings with a name other than **Dynatrace**.

Reply:
Yes, please, providing a custom binding name at runtime is required as we do not control the name in the environment. And even better, it shall work like a `.contains()` (e.g. withName = \*dynatrace\*).

Reply:
+1 but this is probably a separate as the binding support comes from libpak. Rather than add a ton of binding helper methods, it would probably make sense to add something like `bindings.WithFilter(..)` that takes a function callback. If the callback can get the binding information as arguments and return true if it matches or false if it doesn't. This way different buildpacks can implement binding search logic as they require (strict equals vs contains vs starts with vs etc...).

Reply:
TBH, I do not really like the `contains()` approach. This could have strange side effects and I do not see the benefit.
**IF** we want to support different names, I would suggest something like `BP_BINDING_NAME` to point to the name that should be checked. Similar to the handling of `pom.xml` and the like.

@dmikusa WDYT?

But the binding is user provided after all, so why not expect a specific name?

Reply:
In case we can provide the custom name by cf env variable, then the `.equals()` approach is fine for me.

Reply:
While looking at cf-for-k8s in the past we had a cluster using `postgres` in its base URL and the db driver injection we backported was matching for every service... 😁

Reply:
> > > Existence of a Dynatrace SaaS/Managed service is defined as the [VCAP_SERVICES](http://docs.cloudfoundry.org/devguide/deploy-apps/environment-variable.html#VCAP-SERVICES) payload containing a service who's name, label or tag has dynatrace as a substring

That is exactly my problem with that. That is essentially a fancy way of saying "containing the word dynatrace somewhere". Not very specific imho.

> I'd be OK with doing the same here. Given the history of doing this in the CF JBP, I'm not worried about it causing the wrong service to be selected.

Fair point.

> If there's a specific set of circumstances you're worried about though, let's talk more.

I don't think that is necessary. I am not really convinced, but I see your point and it is definitely a solution.

> I don't feel like this should be necessary.

OK.

> If that presents problems, we could circle back on this.

I will update the PR.